ss58-js/mod.ts

```diff
@@ -0,0 +1,89 @@
+import { blake2b } from "../deps/hashes.ts";
+import * as base58 from "../deps/std/encoding/base58.ts";
+
+import { hex } from "../util/mod.ts";
+
+import {
+  ALLOWED_ADDRESS_LENGTHS,
+  ALLOWED_NETWORK_PREFIXES,
+  ALLOWED_PUBLIC_KEY_LENGTHS,
+  CHECKSUM_LENGTH,
+  SS58PRE,
+} from "./constants.ts";
+
+export const encode = (prefix: number, pubKey: string): string => {
+  const pubKeyBytes = hex.decode(pubKey);
+
+  if (!ALLOWED_PUBLIC_KEY_LENGTHS.includes(pubKeyBytes.length)) {
+    throw new Error("Invalid public key length");
```

Comment:
Let's subtype errors.

```ts
import * as U from "../util/mod.ts";

export class InvalidPublicKeyLenError extends ErrorCtor("InvalidPublicKeyLenError") {}
```

This way, their names exist at the static type level.

```ts
const e = new InvalidPublicKeyLenError();

// type is "InvalidPublicKeyLenError"
e.name;

// Also, we can identify the instance type at runtime
if (e instanceof InvalidPublicKeyLenError) {
  // ...
}
```

Make sure to export these error subtypes as well.

Also, we may want to attach additional data.

```diff
import * as U from "../util/mod.ts";

- export class InvalidPublicKeyLenError extends ErrorCtor("InvalidPublicKeyLenError") {}
+ export class InvalidPublicKeyLenError extends ErrorCtor("InvalidPublicKeyLenError") {
+   constructor(readonly actualLen: number) {
+     super();
+   }
+ }
```